### ISSUE TITLE
Startpage.com image search not filtered

### ISSUE THREAD

**Issue by mrusme** [iorate/ublacklist#243]
## Expected Behavior
When I block a site (e.g. www.pinterest.com) I would expect the image search to also be filtered.

## Actual Behavior
The search results are filtered on the regular search, but images still appear on the image search.

## Steps to Reproduce the Problem

1. Search for "logo x" on startpage.com's regular search
2. Block pinterest (you should really do, regardless of this bug)
3. Switch to image search 

## Specifications

- OS: Linux
- Browser: Chromium as well as Firefox
- Other extensions that might interfere: None

**Comment by Semro** [comment]
Not a bug. There is no images support for startpage.com, list of supported features mentioned in [README](https://github.com/iorate/ublacklist#supported-search-engines:~:text=%E2%9C%94%EF%B8%8F-,Startpage,-%E2%9C%94%EF%B8%8F)

**Comment by mrusme** [comment]
@Semro thanks, I overlooked that. Feel free to change this to be a feature-request then! :-)

**Comment by mrusme** [comment]
You could implement it pragmatically and check against `div.image-quick-details > span`:

![screenshot_2022-06-13-095842](https://user-images.githubusercontent.com/151967/173383250-cd2d6e10-3dd7-47c5-9139-a410a0e0305e.png)

It's better than nothing, and it seems they always display the host.domain.tld in there, so for many cases this might work pretty good.

**Comment by mrusme** [comment]
Want to re-open this issue then? :)

**Comment by mrusme** [comment]
@iorate since I couldn't find any documentation, maybe you can tell me: `target` is the element that should be hidden, correct? And also, what are `actionTarget` and `actionStyle`?

Disclaimer: Not a web developer, I don't know what I'm doing.

**Comment by iorate** [comment]
> target is the element that should be hidden

Yes, in most cases.

> what are actionTarget and actionStyle

"Actions" are containers of the "block this site" and "unblock this site" buttons. `actionTarget` designates where actions are inserted, and `actionStyle` designates the style of actions.

---

I found another problem. The image search result page is built on React. It means that elements added by this extension are removed on hydration.
Setting `runAt` to `document_idle` will avoid the problem (ref: #179), but it will worsen the current user experience in the web/video/news search, that is, blocked sites will show up for a moment. I won't accept it.

---

**Conclusion: won't fix** (at least for now)
Image search support would be incomplete, but would either worsen the user experience or require a major rewrite of the code. It's not worth it.
